packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java

```diff
@@ -75,22 +77,27 @@ final class VideoPlayer {
     ExoPlayer exoPlayer = new ExoPlayer.Builder(context).build();
 
     Uri uri = Uri.parse(dataSource);
-    DataSource.Factory dataSourceFactory;
 
-    if (isHTTP(uri)) {
```

Comment:
@zezo357 I am ready to add tests but I have some follow up questions first to make sure I understand the change.

1. Why did you delete the check for `isHttp(...)` and use `new DefaultDataSource.Factory(context, httpDataSourceFactory)` by default versus `httpDataSourceFactory.setDefaultRequestProperties(httpHeaders)`? What happens if the uri is not HTTP?
2. As a follow up to (1), was the fix for passing mu38 files to add the `.setUserAgent(httpHeaders.get(USER_AGENT))` line if `USER_AGENT` is present the HTTP headers?

Reply:
1-If not http all functions will work and it's the main fix since when playing a m3u8 file it's not http 


2- user agent should be added to the http headers If the user wants it (it's needed because some hosts do not allow the default user agent)